File: Basic/Day16_3Intent/src/com/example/day16_3intent/SecondActivity.java
package com.example.day16_3intent;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

public class SecondActivity extends Activity {
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.second);// 현재java파일과 연결할 layout파일 등록

		Button b1 = (Button) findViewById(R.id.button1);
		b1.setOnClickListener(new OnClickListener() {
			public void onClick(View v) {
				Intent intent = new Intent(getApplicationContext(), ThirdActivity.class);
				startActivity(intent);
			}
		});
		Button b2 = (Button) findViewById(R.id.button2);
		b2.setOnClickListener(new OnClickListener() {
			public void onClick(View v) {
				Intent intent = new Intent(getApplicationContext(), // 현재화면의
																	// 제어권자
						MainActivity.class);// 다음 넘어갈 화면
				finish();
				;// 현재 화면을 닫아줌
				startActivity(intent);
			}
		});

	}// end of onCreate
}// end of class

File: StudioProject/FirstApp/app/src/main/kotlin/skim/dev/kr/firstapp/SecondActivity.kt
/**
 * Created by suyoung on 2018. 2. 25..
 */
package skim.dev.kr.firstapp

import android.app.Activity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.util.TypedValue
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.*

class SecondActivity : Activity(), AnkoLogger {

    override val loggerTag: String
        get() = "MyLog"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val manager : LinearLayoutManager = LinearLayoutManager(this)
        view.layoutManager = manager

        val adapter : KCityAdapter = KCityAdapter()
        view.adapter = adapter

        val cities = listOf("Seoul", "Tyoko", "Mountaion View", "Singapore")
//        selector(title="Select City", items=cities) { dlg, selection ->
//            toast("You selected ${cities[selection]}")
//        }

//        alert(title="Message", message="Let's learn Kotlin!") {
//            positiveButton("Yes") {
//                toast("Yay!")
//            }
//
//            negativeButton("No") {
//                longToast("No way...")
//            }
//        }.show()

//        val pd = progressDialog(title = "File Download", message = "Downloaing ...")
//        pd.show()
//        pd.progress = 50
//        indeterminateProgressDialog(message="Please wait..").show()

//        val intent = Intent(this.ctx, KDetailActivity::class.java)
//        intent.putExtra("id", 150L)
//        intent.putExtra("title", "Awesome item")
//        intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
//        startActivity(intent)
//        val intent = intentFor<KDetailActivity>(
//                "id" to 150L, "title" to "Awesome item"
//        ).noHistory()
//        startActivity(intent)
        startActivity<KDetailActivity>("id" to 150L, "title" to "Awesome item")
//        sendSMS(number = "01096220213", text = "Hello")

        val dpInPixel = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 100f, resources.displayMetrics)
        val spInPixel = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, 16f, resources.displayMetrics)
        info(dpInPixel.toString() + "  : " + dip(100))
        info(spInPixel.toString() + "  : " + sp(16).toFloat())

        verbose("??verbose")
        debug("??!!!!!")
        info("????")
        warn("??warn")
        error("??error")
        wtf("??wtf")

    }

}